File: 1291.go
import "slices"

func sequentialDigits(low int, high int) []int {
	ans := make([]int, 0)
	q := make([]int, 0)
	for i := 1; i <= 9; i++ {
		q = append(q, i)
	}
	for len(q) > 0 {
		c := q[0]
		q = q[1:]
		if c >= low && c <= high {
			ans = append(ans, c)
		} else if c > high {
			continue
		}
		d := c % 10
		if d != 9 {
			q = append(q, c*10+d+1)
		}
	}
	slices.Sort(ans)
	return ans
}

File: 202607/1291.cpp
class Solution {
  using ll = long long;

public:
  vector<int> sequentialDigits(int low, int high) {
    vector<int> ans;
    function<void(ll)> f = [&](ll curr) {
      if (curr > high)
        return;
      if (low <= curr && curr <= high) {
        ans.push_back(curr);
      }
      int last_digit = curr % 10;
      if (last_digit != 9) {
        ++last_digit;
        f(curr * 10 + last_digit);
      }
    };
    for (int i = 1; i < 10; ++i) {
      f(i);
    }
    sort(begin(ans), end(ans));
    return ans;
  }
};
